securedrop/source_app/__init__.py

```diff
@@ -124,11 +124,23 @@ def setup_g() -> Optional[werkzeug.Response]:
         if 'expires' in session and datetime.utcnow() >= session['expires']:
             msg = render_template('session_timeout.html')
 
+            # Before the session is cleared, record the fact that
+            # the user was in the codename generation flow or logged in
+            generate_flow_record = session.get('generate_flow_record', was_in_generate_flow())
+            login_record = session.get('login_record', logged_in())
```

Comment:
I debated whether I should make names of these properties long and descriptive or relatively short like existing ones. Not entirely happy with names I came up, so I'd welcome any suggestions.

Reply:
What about using a single session property to determine whether to flash the expiration message? 
```python
            [...]
            show_expiration_message = any([
                session.get('show_expiration_message'),
                logged_in(),
                was_in_generate_flow(),
            ])

            # clear the session after we render the message so it's localized
            session.clear()

            session['show_expiration_message'] = show_expiration_message
            [...]
```

Reply:
Makes sense! I made the change and updated comments accordingly.

Reply:
Cool. Thanks.